### ISSUE TITLE
Nano branch OpenCV 4.1.0 compatibility issues.

### ISSUE THREAD

**Issue by courageon** [hoangthien94/ORB_SLAM2_CUDA#14]
Hey folks! I'm trying to containerize this project specifically for the Jetson Nano following the instructions and changes @thien94 has awesomely created for us. I'm having issues specifically with the types that went missing between OpenCV versions 3->4:
https://github.com/opencv/opencv/wiki/Opencv4

PnPsolver.cc and PnPsolver.h both reference the "old-style" CvMat structure and functions associated which do not exist in OpenCV 4.1.0. There's also some svd solver functions and enum's that are missing.

There are also some color conversion enums (like CV_RGB2GRAY) in Tracking.cc that don't exist in OpenCV > 4.0.1

So I'm just wondering if there's some magical include I can add to these files to provide compatibility? Or are there some changes possibly missing to get this project fully under OpenCV > 4.0.1? I just want to make sure I'm not reproducing effort, or doing something completely unnecessary!

Thanks!

**Comment by thien94** [comment]
Hi @courageon , I do see a patch for the original ORB-SLAM2 that seems to suggest that you can do just that: changing the includes and some other fixes to upgrade opencv3->4. See here: https://github.com/raulmur/ORB_SLAM2/pull/790
Hope it works for you.

**Comment by courageon** [comment]
Awesome! Looks like there's a couple of includes to add the c-types into the headers that needed them, and that worked!

Going to try to get it with opencv 4.1.2 now since I downgraded to 4.0.0 for that build!
Edit: Runs with 4.1.2 within the container.

Any suggestions on how to get it working with a usb camera or CSI interface camera, like the raspberry pi camera? I can start a new issue if need be, since this would be off topic now. 
Edit: Found the associated config yaml! Nice! Needed to modify tx1.cpp to use capture index for usb camera, but looks like it defaults to using the CSI interface! :+1:
